### PR TITLE
save train/val/test pk files when raw total provided

### PR DESCRIPTION
**Pull request by pzhanggit**
- [x] Save *train.pkl, *validate.pkl and *test.pkl files used in training when total dataset is provided for consistency in continuing training, comparison etc.
- [x] remove "raw" since not needed

### CODE REVIEW

**Comment by pzhanggit** [comment]
@allaffa  See this PR for separate pkl files

File: hydragnn/preprocess/load_data.py [hunk 1]
```diff
@@ -171,8 +175,27 @@ def transform_raw_data_to_serialized(config):
     _, rank = get_comm_size_and_rank()
 
     if rank == 0:
-        loader = RawDataLoader(config)
+        loader = RawDataLoader(config["Dataset"])
         loader.load_raw_data()
+        if len(loader.serial_data_name_list) == 1 and "total" in loader.path_dictionary:
```
Comment (streeve) [inline hydragnn/preprocess/load_data.py:180]: Let's move everything in this `if` into a separate function

File: hydragnn/preprocess/load_data.py [hunk 2]
```diff
@@ -76,22 +82,24 @@ def create_dataloaders(trainset, valset, testset, batch_size):
     return train_loader, val_loader, test_loader
 
 
-def split_dataset(
-    dataset_list: [],
-    dataset_names: [],
-    batch_size: int,
-    perc_train: float,
-):
+def split_dataset(dataset: [], perc_train: float):
```
Comment (streeve) [inline hydragnn/preprocess/load_data.py:85]: I like this cleanup, but you should add an issue for cleaning up the function names

Comment (allaffa) [reply hydragnn/preprocess/load_data.py:85]: @streeve 
Do you mean open an issue on GutHub and link it to this PR?

Comment (streeve) [reply hydragnn/preprocess/load_data.py:85]: That was my intent, yes. I should have really said clean up naming and intent of each function - it's always hard for me to remember what each function here actually does

**Comment by pzhanggit** [comment]
I updated the script after our discussion. Let me know what you think @streeve

**Review by allaffa — APPROVED** [review]

**Pull request merged**